src/main/java/teammates/common/util/Url.java

```diff
@@ -107,34 +65,38 @@ public String get(String parameterName) {
      * </ul>
      */
     public static String addParamToUrl(String url, String key, String value) {
-        if (key == null || value == null) {
-            // return the url if any of the key or the value is null
+        if (key == null || key.isEmpty() || value == null || value.isEmpty()
+             || url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
+            // return the url if any of the key or the value is null or empty
+            // or if the key is already included in the url
             return url;
         }
-        if (url.contains("?" + key + "=") || url.contains("&" + key + "=")) {
-            // return the url if the key is already included in the url
-            return url;
-        }
-        url += url.indexOf('?') >= 0 ? '&' : '?';
-        url += key + "=" + Sanitizer.sanitizeForUri(value);
-        return url;
+        return url + (url.contains("?") ? "&" : "?") + key + "=" + Sanitizer.sanitizeForUri(value);
     }
 
     public static String trimTrailingSlash(String url) {
         return url.trim().replaceAll("/(?=$)", "");
     }
+    
+    public static String getRelativePath(String url) {
+        try {
+            return new Url(url).toString();
+        } catch (AssertionError e) {
+            return url;
+        }
+    }
```

Comment:
Unfortunately yes, the only use case of this method is getting the relative path of a `WebElement`'s `href` attribute. In Firefox the behavior is fixed: Selenium returns the full URL even though the attribute value is relative. Not sure about Chrome, but [looks like it's different](https://code.google.com/p/selenium/issues/detail?id=1824). If we want to support _only_ Firefox, then yes I can use the simple `return new Url(url).getRelativePath();` method.